.travis.yml

```diff
@@ -0,0 +1,49 @@
+language: cpp
+cache: ccache
+sudo: required
+dist: xenial
+
+addons:
+  apt:
+    sources:
+    - ubuntu-toolchain-r-test
+    packages:
+    - g++-8
+    - clang-3.8
+    - build-essential
+    - pkg-config
+    - cmake
+    - libpython-dev python-numpy
+    - libboost-all-dev
+
+before_install:
+  - if [ "$TRAVIS_OS_NAME" == "osx" ]; then brew update          ; fi
+
+script:
+  - bash .travis.sh
+
+env:
+  global:
+    - MAKEFLAGS="-j 2"
+    - CCACHE_SLOPPINESS=pch_defines,time_macros
+
+matrix:
+  include:
+  - compiler: gcc
```

Comment:
As per the travis docs, the default gcc version is 5.4. Should we increase this gcc 6.3 since it seems to be a bit faster as per [this](https://www.phoronix.com/scan.php?page=article&item=gcc-7-january&num=2). We can always add gcc 5.4 with examples off as an additional build matrix element.